### PR TITLE
Improve query interface for get recordings page

### PR DESCRIPTION
**Pull request by arthurmcgregor**
Significant changes to the get recordings page.

- Changed layout of columns to be more responsive (#76)
- Changed terms for date selection. Before date and after date become From Date and To Date (which makes more sense to me)
- Replaced longer than, shorter than with duration range slider
- Duration range slider initially has range from 0 to 100 s, however clicking the arrow will increase the max range by 100s (up to 600s)
- Date and duration automatically create new conditions (and remove old conditions) when their input changes. _I am assuming you only ever want one set of bounds for each condition_
- Conditions have been hidden. Now that they always reflect the dates and duration boxes you do not need to see them or delete old conditions.
- Shifted device and tag selectors to the top of the page. _I am assuming these are more commonly used_
- Rearranged limit and offset to be closer to table
- Rearranged other buttons
- Added table-responsive class to table which means it becomes scroll-able on small screens (better than alternative)

So hopefully that all works and is an improvement.

A few questions (because I don't really know how this is actually used!)
- Are devices and tags the most commonly used query fields, or should something else be at the top of the page?
- Do you ever want to filter by (devices || tags) && (duration || date)? Because 'get all' only does (devices || tags) and 'get all from conditions' only does (date || duration). I feel these two could be combined into one 'Search' button or 'Filter' or 'Get recordings'
- Do you ever want to choose several devices? ie a checklist rather than a drop down list
- Ditto for tags

Screenshots
Desktop size - before
desktop1
Desktop size - after
desktop2
Mobile size - before
mobile1
Mobile size - after
mobile2

### CODE REVIEW

**Comment by mjs** [comment]
This is excellent. Thanks! Answers to your questions:

**Are devices and tags the most commonly used query fields, or should something else be at the top of the page?**

The device and tags fields are the most common ones.

**Do you ever want to filter by (devices || tags) && (duration || date)? Because 'get all' only does (devices || tags) and 'get all from conditions' only does (date || duration). I feel these two could be combined into one 'Search' button or 'Filter' or 'Get recordings'**

There should be just one button. I like "Search" as the name for it.

**Do you ever want to choose several devices? ie a checklist rather than a drop down list**

Not really, but we would like to be able to filter by "device group". We'll have to figure out how to query the API for that.

**Ditto for tags**

The field called "Tags" should actually be called "Tag type" or "Tag class". The current combinations in the drop down are all we need right now.

At some stage we need to add searching by tag (i.e. animal). That would be nice to do as a series of checkboxes.

See my email too for more UI suggestions. No need to do them all in one PR though. This is a great start.

I'll leave the code review for @CameronRP .

File: public/javascripts/getRecordings.js [hunk 1]
```diff
@@ -39,26 +40,40 @@ window.onload = function() {
...
   conditions[id] = sequelizeCondition;
   updateConditions();
 }
 
 // Removes a Sequelize condition with the given ID.
 function deleteCondition(id) {
-  console.log('Delete condition: ', id);
+  // console.log('Delete condition: ', id);
   delete conditions[id];
   updateConditions();
```
Comment (CameronRP) [inline public/javascripts/getRecordings.js]: Removing the conditions element is causing `updateConditions()` function to error. This then causes the `addToDate()` and other similar functions to error causing unwanted behavior. Simple fix should be to delete this function and all it's references.

**Review by CameronRP — CHANGES_REQUESTED** [review]
Nice work! Looks much better. Just one little fix.

File: public/javascripts/getRecordings.js (hunk 1 above)
Comment (arthurmcgregor) [reply public/javascripts/getRecordings.js]: Great thanks. I initially kept it in there for debugging.... but totally gone now :-)

**Comment by arthurmcgregor** [comment]
Thanks for your email and feedback @mjs. Will look to incorporate those ideas as well...

**Comment by mjs** [comment]
@arthurmcgregor Did you want to land this and then propose more PRs with some of the other suggestions?

**Comment by arthurmcgregor** [comment]
@mjs Yup will do :-)

**Review by CameronRP — APPROVED** [review]

**Comment by CameronRP** [comment]
@arthurmcgregor Code is now live on the test server :) https://browse-test.cacophony.org.nz/get_recordings

**Comment by arthurmcgregor** [comment]
Yay! Now I can see how well it works on an actual mobile phone...

Plus summarised all of these ideas in #81

**Comment by mjs** [comment]
@arthurmcgregor it's already so much better - thank you! Looking forward to seeing what else you come up with.

**Pull request merged**